traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go

```diff
@@ -263,7 +243,409 @@ func (dss *TODeliveryServiceServer) Delete(db *sqlx.DB, user auth.CurrentUser) (
+			dtos := map[string]interface{}{"id":dsId, "server":servers[i]}
+			resultRows, err := tx.NamedQuery(insertIdsQuery(), dtos)
+			if err != nil {
+				if pqErr, ok := err.(*pq.Error); ok {
+					err, eType := dbhelpers.ParsePQUniqueConstraintError(pqErr)
+					log.Error.Printf("could not begin transaction: %v", err)
+					if eType == tc.DataConflictError {
+						handleErrs(http.StatusInternalServerError, err)
+						return
+					}
+					handleErrs(http.StatusInternalServerError, err)
+					return
+				}
+				log.Errorf("received non pq error: %++v from create execution", err)
+				return
+			}
+			respServers = append(respServers, servers[i])
+			resultRows.Next()
+			i++
+			defer resultRows.Close()
+		}
+
+		err = tx.Commit()
+		if err != nil {
+			log.Errorln("Could not commit transaction: ", err)
+			return
+		}
+		rollbackTransaction = false
+		resAlerts := []tc.Alert{ tc.Alert{"sever assignement","success"}}
+		repRes := tc.DSSReplaceResponse{resAlerts, tc.DSSMapResponse{*dsId,*payload.Replace,respServers }}
+
+		// marshal the results to the response stream
+		respBts, err := json.Marshal(repRes)
+		if err != nil {
+			log.Errorln("Could not marshal the response as expected: ", err)
+			handleErrs(http.StatusInternalServerError, err)
+			return
+		}
+
+		w.Header().Set(tc.ContentType, tc.ApplicationJson)
+		fmt.Fprintf(w, "%s", respBts)
+		return
+	}
+}
+
+
+
+
+type TODeliveryServiceServers tc.DeliveryServiceServers
+
+var serversRef = TODeliveryServiceServers(tc.DeliveryServiceServers{})
+
+func GetServersRef() *TODeliveryServiceServers {
+	return &serversRef
+}
+
+// api/1.1/deliveryservices/{xml_id}/servers
+func GetCreateHandler( db *sqlx.DB ) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		handleErrs := tc.GetHandleErrorsFunc(w, r)
+
+		// find the delivery service Id dsId matching the xml_id
+		params, err := api.GetCombinedParams(r)
+		if err != nil {
+			log.Errorf("unable to get parameters from request: %s", err)
+			handleErrs(http.StatusInternalServerError, err)
+		}
+
+		xmlId, ok := params["xml_id"]
+		if !ok {
+			log.Errorf("unable to get xml_id parameter from request: %s", err)
+			handleErrs(http.StatusInternalServerError, err)
+			return
+		}
+
+		ctx := r.Context()
+		user, err := auth.GetCurrentUser(ctx)
+		if err != nil {
+			log.Errorf("unable to retrieve current user from context: %s", err)
+			handleErrs(http.StatusInternalServerError, err)
+			return
+		}
+
+		// if the object has tenancy enabled, check that user is able to access the tenant
+		// check user tenancy access to this resource.
+		hasAccess, err, apiStatus := tenant.HasTenant(*user, xmlId, db)
+		if !hasAccess {
+			switch apiStatus {
+			case tc.SystemError:
+				handleErrs(http.StatusInternalServerError, err)
+				return
+			case tc.DataMissingError:
+				handleErrs(http.StatusBadRequest, err)
+				return
+			case tc.ForbiddenError:
+				handleErrs(http.StatusForbidden, err)
+				return
+			}
```

Comment:
The `switch` needs a `default`, which writes `http.StatusInternalServerError` and logs the unknown API status. Otherwise, if someone adds a new API status, this will keep going and provide access when it shouldn't.